java/code/src/com/redhat/rhn/manager/system/entitling/SystemUnentitler.java

```diff
@@ -73,8 +82,8 @@ public void removeServerEntitlement(Server server, Entitlement ent) {
             unentitleServer(server, ent);
         }
 
-        if (EntitlementManager.MONITORING.equals(ent)) {
-            server.asMinionServer().ifPresent(s -> {
+        server.asMinionServer().ifPresent(s -> {
+            if (EntitlementManager.MONITORING.equals(ent)) {
                 ServerGroupManager.getInstance().updatePillarAfterGroupUpdateForServers(Arrays.asList(s));
```

Comment:
line 87 " ServerGroupManager.getInstance().updatePillarAfterGroupUpdateForServers(Arrays.asList(s)); "
should be executed despite the minion has a monitoring entitlement or not. So it should be outside this 'if' statement

Reply:
Fixed